source/retryable-writes/retryable-writes.rst

```diff
@@ -212,24 +212,34 @@ Determining Retryable Errors
 
 When connected to a MongoDB instance that supports retryable writes (versions 3.6+),
 the driver MUST treat all errors with the RetryableWriteError label as retryable.
-This label might be added to an error in a variety of ways:
+This error label can be found in the top-level "errorLabels" field of the error.
+With a WriteConcernError response, the top level document or the
+WriteConcernError document may contain the RetryableWriteError error label.
+
+The RetryableWriteError label might be added to an error in a variety of ways:
 
 When the driver encounters a network error communicating with any server version
 that supports retryable writes, it MUST add a RetryableWriteError label to that
-error.
+error if the MongoClient performing the operation has the retryWrites
+configuration option set to true.
 
 For server versions 4.4 and newer, MongoDB will add a RetryableWriteError label to
 errors or server responses that it considers retryable before returning them to the
 driver. As new server versions are released, the errors that are labeled with the
 RetryableWriteError label may change. When receiving a command result
 with an error from a 4.4+ server that supports retryable writes, the driver
-MUST NOT add a RetryableWriteError label to that error under any condition.   
-Note: With a WriteConcernError response the top level document or the 
-WriteConcernError document may contain the RetryableWriteError error label.
+MUST NOT add a RetryableWriteError label to that error under any condition.
+
+During a retryable write operation on a sharded cluster, mongos may retry the
```

Comment:
Is there a test case for this scenario?

Reply:
I believe the [errorLabel test cases](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/tests/insertOne-errorLabels.yml) cover this scenario -- they run on sharded clusters and test the case where the server returns an error label and when it doesn't, which is essentially the behavior described here. Does that seem reasonable?

Reply:
I agree, the behavior seems covered by this test.